docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,172 @@
+# Downward APIs for resource limits and requests
+
+## Background
+Currently downward APIs (via environment variables and volume plugin) only support exposing a Pod's name, namespace, annotations, labels and its IP. This document explains the need and design to extend them to expose resources (e.g. cpu, memory, storage) limits and requests.
+
+## Motivation
+By exposing information about computing resource requests & limits (e.g. cpu, memory, storage) into pods, image/application authors can be aware of the computing environment in which the pods run, and could modify container images accordingly.
+
+## Design
+This is mostly driven by the discussion in https://github.com/kubernetes/kubernetes/issues/9473 with some additonal stuff.
+
+### Downward APIs via environment variables and volume plugin
+
+#### APIs with full json path selectors
+
+Full json path selectors specify the complete path to the resources limits and request relative to pod spec.
+
+##### Environment variables
+
+| Name | Selectors |
+| ---- | ------------------- |
+| CPU_LIMIT | .spec.containers[?(@.name=="container-name")].resources.limits.cpu |
+| MEMORY_LIMIT | .spec.containers[?(@.name=="container-name")].resources.limits.memory |
+| STORAGE_LIMIT | .spec.containers[?(@.name=="container-name")].resources.limits.storage |
+| CPU_REQUEST | .spec.containers[?(@.name=="container-name")].resources.requests.cpu |
+| MEMORY_REQUEST | .spec.containers[?(@.name=="container-name")].resources.requests.memory |
+| STORAGE_REQUEST | .spec.containers[?(@.name=="container-name")].resources.requests.storage |
+
+Since environment variables are container scoped, the container name should be specified as part of the selector.
+
+##### Volume plugin
+
+| Path | Selectors | Resulting path|
```

Comment:
I do not understand what this table is supposed to be, and I think it may represent a mismatch -- can you introduce the table and say what it's supposed to mean so I can decide on whether a mismatch is present?